src/main/java/gyro/google/kms/CryptoKeyResource.java

```diff
@@ -0,0 +1,382 @@
+        return purpose;
+    }
+
+    public void setPurpose(CryptoKey.CryptoKeyPurpose purpose) {
+        this.purpose = purpose;
+    }
+
+    /**
+     * The template describing settings for new crypto key versions. (Required)
+     *
+     * @subresource gyro.google.kms.CryptoKeyVersionTemplate
+     */
+    @Updatable
+    @Required
+    public CryptoKeyVersionTemplate getCryptoKeyVersionTemplate() {
+        return cryptoKeyVersionTemplate;
+    }
+
+    public void setCryptoKeyVersionTemplate(CryptoKeyVersionTemplate cryptoKeyVersionTemplate) {
+        this.cryptoKeyVersionTemplate = cryptoKeyVersionTemplate;
+    }
+
+    /**
+     * The labels of the crypto key.
+     */
+    @Updatable
+    public Map<String, String> getLabels() {
+        if (labels == null) {
+            labels = new HashMap<>();
+        }
+
+        return labels;
+    }
+
+    public void setLabels(Map<String, String> labels) {
+        this.labels = labels;
+    }
+
+    /**
+     * The ID of the primary key version.
+     */
+    @Updatable
+    public String getPrimaryKeyVersionId() {
+        return primaryKeyVersionId;
+    }
+
+    public void setPrimaryKeyVersionId(String primaryKeyVersionId) {
+        this.primaryKeyVersionId = primaryKeyVersionId;
+    }
+
+    /**
+     * The ID of the crypto key.
+     */
+    @Output
+    @Id
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * The list of versions of the crypto key.
+     */
+    @Output
+    public List<String> getVersions() {
+        if (versions == null) {
+            versions = new ArrayList<>();
+        }
+        return versions;
+    }
+
+    public void setVersions(List<String> versions) {
+        this.versions = versions;
+    }
+
+    @Override
+    public void copyFrom(CryptoKey model) throws Exception {
+        setId(model.getName());
+        setPurpose(model.getPurpose());
+        setKeyRing(findById(KeyRingResource.class, String.join("/", Arrays.copyOfRange(getId().split("/"), 0, 6))));
+        setName(getNameFromId());
+
+        if (model.hasNextRotationTime()) {
+            DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy");
+            Date date = new Date(model.getNextRotationTime().getSeconds() * 1000L);
+            setNextRotationDate(dateFormat.format(date));
+        }
+
+        if (model.hasRotationPeriod()) {
+            setRotationPeriod(model.getRotationPeriod().getSeconds() / 86400);
+        }
+
+        if (model.getVersionTemplate() != null) {
+            CryptoKeyVersionTemplate cryptoKeyVersionTemplate = newSubresource(CryptoKeyVersionTemplate.class);
+            cryptoKeyVersionTemplate.copyFrom(model.getVersionTemplate());
+            setCryptoKeyVersionTemplate(cryptoKeyVersionTemplate);
+        }
+
+        if (model.hasPrimary()) {
+            setPrimaryKeyVersionId(model.getPrimary().getName().split("/")[9]);
+        }
```

Comment:
else clause ???

Reply:
I can remove the null check for the `getVersionTemplate` since its s required field and it can't be null. I don't exactly remember why I added the null check, but I'll keep you updated in case I can't.

For the others, I don't think we require else clauses since the fields are null values anyways.